src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -4617,6 +4640,14 @@ private IAsyncResult BeginConnectEx(EndPoint remoteEP, bool flowContext, AsyncCa
             EndPoint endPointSnapshot = remoteEP;
             Internals.SocketAddress socketAddress = Serialize(ref endPointSnapshot);
 
+            if (SocketsTelemetry.Log.IsEnabled())
+            {
+                SocketsTelemetry.Log.ConnectStart(socketAddress);
+
+                // Ignore flowContext when using Telemetry to avoid losing Activity tracking
+                flowContext = true;
```

Comment:
How does this forces `ConnectOverlappedAsyncResult` to capture the context?
It's captured in `ContextAwareResult.CaptureOrComplete` which does it only if `_flags` have `CaptureContext` but that seems to be off by default (base ctor defaults parameter `forceCaptureContext` to `false`). Or I'm missing something fundamental here...

Reply:
Forcing `flowContext` to true here means we will alsways call `asyncResult.StartPostingAsyncOp`, thus `FinishPostingAsyncOp` won't be a no-op.
To my knowledge, `ContextAwareResult` will always capture the context if a callback is given (and Start/Finish were called and we're not completing synchronously).

Comment in ContextAwareResult:

> Setting forceCaptureContext enables the ContextCopy property even when a null callback is specified.  (The context is always captured if a callback is given.)

Reply:
I see now, thanks for the explanation :+1: